### PR TITLE
WIP: Add Karmashapes source file

### PR DESCRIPTION
**Pull request by stepps00**
WIP. Will fix https://github.com/whosonfirst/whosonfirst-sources/issues/201

This PR adds a source json file for the Karmashapes dataset, which links to the generic CC-BY license. This PR also includes a karmashapes_remarks markdown file, based off similar remarks files (mesoshapes, quattroshapes, etc) already in this repo that contain the input source names and links used in the project. The LGA dataset was not included since it was only used as a reference.

If the info in this PR looks good, I'll run the makefile to update supplemental files and merge shortly.

### CODE REVIEW

**Comment by stepps00** [comment]
cc @justinelliotmeyers

File: sources/karmashapes_remarks.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+### Karmashapes
+
+Properties with the `ks` prefix are from Karmashapes.
+
+**Examples:** `ks:id` for Karmashapes identifier concordance and `ks:name` for place names. 
+
+Karmashapes is included in Who's On First's administrative data repo in India. Who's on First doesn't always use the Karmashapes geometry, so review that in `src:geom`. When we do, some Karmashapes features source from friendly CC0 or Public Domain licensed projects, and others from a stricter CC-BY project.
+
+Please note the following with regard to Quattroshapes features that require attribution:
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:9]: `Karmashapes` (from Quattroshapes)

File: sources/karmashapes.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    "license_type": "CC-BY", 
+    "name": "karmashapes", 
+    "prefix": "ks", 
+    "remarks": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/karmashapes_remarks.md", 
+    "src:via": [
+        {
+            "context": "India", 
+            "source_link": "https://geonames.nga.mil/geonames/GNSHome/welcome.html", 
+            "source_name": "National Geospatial Intelligence Agency (NGA)", 
+            "source_note": ""
```
Comment (nvkelso) [inline sources/karmashapes.json:18]: Is this where we'd indicate the license information? Right now that's only in the remarks file.

Comment (nvkelso) [reply sources/karmashapes.json:18]: Might also note which NGA layer, eg "global populated places"?

File: sources/karmashapes.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            "context": "India", 
+            "source_link": "https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download", 
+            "source_name": "Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN)", 
+            "source_note": ""
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://geosadak-pmgsy.nic.in/opendata/", 
+            "source_name": "Pradhan Mantri Gram Sadak Yojana (PMGSY)", 
+            "source_note": ""
```
Comment (nvkelso) [inline sources/karmashapes.json:30]: Consider noting license.

Their required credit is: `Ministry of Rural Development, 2022. PMGSY`

The data layer is called "Habitations".

File: sources/karmashapes.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://ghsl.jrc.ec.europa.eu/ghs_stat_ucdb2015mt_r2019a.php", 
+            "source_name": "Global Human Settlement Layer (GHSL)", 
+            "source_note": ""
```
Comment (nvkelso) [inline sources/karmashapes.json:36]: The data layer is `Urban Centre Database UCDB R2019A`?

Agree CC-BY 4.0 via https://commission.europa.eu/legal-notice_en.

File: sources/karmashapes.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://www.arcgis.com/apps/instant/media/index.html?appid=fc92d38533d440078f17678ebc20e8e2", 
+            "source_name": "ESRI’s Sentinel-2 10m Land Use/Land Cover", 
+            "source_note": ""
```
Comment (nvkelso) [inline sources/karmashapes.json:42]: Agree CC-BY 4.0 per https://www.arcgis.com/home/item.html?id=d3da5dd386d140cf93fc9ecbf8da5e31

File: sources/karmashapes.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://download.geoservice.dlr.de/WSF_EVO/", 
+            "source_name": "World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)", 
+            "source_note": ""
```
Comment (nvkelso) [inline sources/karmashapes.json:48]: Agree CC-BY 4.0

File: sources/karmashapes.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            "context": "India", 
+            "source_link": "https://geonames.nga.mil/geonames/GNSHome/welcome.html", 
+            "source_name": "National Geospatial Intelligence Agency (NGA)", 
+            "source_note": ""
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download", 
+            "source_name": "Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN)", 
+            "source_note": ""
```
[the remark this answers is not in the document]
Comment (nvkelso) [reply sources/karmashapes.json:24]: Agree this is CC-BY equivalent per https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/metadata

File: sources/karmashapes_remarks.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+* National Geospatial Intelligence Agency (NGA)
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
+	* License: Public Domain
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
+	* URL: https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download
+	* License: CC-BY
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:16]: If this has a version, then also include that.

File: sources/karmashapes_remarks.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+Properties with the `ks` prefix are from Karmashapes.
+
+**Examples:** `ks:id` for Karmashapes identifier concordance and `ks:name` for place names. 
+
+Karmashapes is included in Who's On First's administrative data repo in India. Who's on First doesn't always use the Karmashapes geometry, so review that in `src:geom`. When we do, some Karmashapes features source from friendly CC0 or Public Domain licensed projects, and others from a stricter CC-BY project.
+
+Please note the following with regard to Quattroshapes features that require attribution:
+
+* National Geospatial Intelligence Agency (NGA)
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:12]: Should these be formatted source, then subitem for which layer (add)?

File: sources/karmashapes_remarks.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+Properties with the `ks` prefix are from Karmashapes.
+
+**Examples:** `ks:id` for Karmashapes identifier concordance and `ks:name` for place names. 
+
+Karmashapes is included in Who's On First's administrative data repo in India. Who's on First doesn't always use the Karmashapes geometry, so review that in `src:geom`. When we do, some Karmashapes features source from friendly CC0 or Public Domain licensed projects, and others from a stricter CC-BY project.
+
+Please note the following with regard to Quattroshapes features that require attribution:
+
+* National Geospatial Intelligence Agency (NGA)
```
[the remark this answers is not in the document]
Comment (nvkelso) [reply sources/karmashapes_remarks.md:11]: Might also note which NGA layer, eg "global populated places"?

File: sources/karmashapes_remarks.md (hunk 8 above)
Comment (nvkelso) [reply sources/karmashapes_remarks.md:16]: Consider saying in plain english the **data layer** from these organizations as a new sub-bullet? In this case: 

- India Village-Level Geospatial Socio-Economic Data Set, v1 (1991, 2001)

This looks to be `CC-BY equivalent` per https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/metadata

File: sources/karmashapes_remarks.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
+	* License: Public Domain
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
+	* URL: https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download
+	* License: CC-BY
+* Pradhan Mantri Gram Sadak Yojana (PMGSY);
+	* URL: https://geosadak-pmgsy.nic.in/opendata/
+	* License: CC-BY (equivalent)
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:19]: The data layer is called "Habitations".

File: sources/karmashapes_remarks.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+Please note the following with regard to Quattroshapes features that require attribution:
+
+* National Geospatial Intelligence Agency (NGA)
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
+	* License: Public Domain
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
+	* URL: https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download
+	* License: CC-BY
+* Pradhan Mantri Gram Sadak Yojana (PMGSY);
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:17]: Change source name to `Ministry of Rural Development, 2022. PMGSY`

Then the dataset name (layer) is `Pradhan Mantri Gram Sadak Yojana (PMGSY)` as new sub-bullet. That  roughly translates to "Prime Ministers Rural Road Schema" to connect more villages with all weather roads.

File: sources/karmashapes_remarks.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+	* URL: https://geosadak-pmgsy.nic.in/opendata/
+	* License: CC-BY (equivalent)
+* Global Human Settlement Layer (GHSL);
+	* URL: https://ghsl.jrc.ec.europa.eu/ghs\_stat\_ucdb2015mt\_r2019a.php
+	* License: CC-BY
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:22]: Is the data layer `Urban Centre Database UCDB R2019A` or `Global Human Settlement Layer (GHSL)`? Are they the same thing?

Agree `CC-BY 4.0` via https://commission.europa.eu/legal-notice_en, but indicate the version here too.

File: sources/karmashapes_remarks.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+* National Geospatial Intelligence Agency (NGA)
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
+	* License: Public Domain
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
+	* URL: https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download
+	* License: CC-BY
+* Pradhan Mantri Gram Sadak Yojana (PMGSY);
+	* URL: https://geosadak-pmgsy.nic.in/opendata/
+	* License: CC-BY (equivalent)
+* Global Human Settlement Layer (GHSL);
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:20]: The credit should read instead: `© European Union, 1995-2023 via the Joint Regional Commission`.

File: sources/karmashapes_remarks.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+	* URL: https://ghsl.jrc.ec.europa.eu/ghs\_stat\_ucdb2015mt\_r2019a.php
+	* License: CC-BY
+* ESRI’s Sentinel-2 10m Land Use/Land Cover;
+	* URL: https://www.arcgis.com/apps/instant/media/index.html?appid=fc92d38533d440078f17678ebc20e8e2
+	* License: CC-BY
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:25]: Agree CC-BY 4.0 per https://www.arcgis.com/home/item.html?id=d3da5dd386d140cf93fc9ecbf8da5e31

File: sources/karmashapes_remarks.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+* World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)
+	* URL: https://download.geoservice.dlr.de/WSF\_EVO/
+	* License: CC-BY
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:28]: Agree CC-BY 4.0

File: sources/karmashapes_remarks.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
+	* URL: https://sedac.ciesin.columbia.edu/data/set/india-india-village-level-geospatial-socio-econ-1991-2001/data-download
+	* License: CC-BY
+* Pradhan Mantri Gram Sadak Yojana (PMGSY);
+	* URL: https://geosadak-pmgsy.nic.in/opendata/
+	* License: CC-BY (equivalent)
+* Global Human Settlement Layer (GHSL);
+	* URL: https://ghsl.jrc.ec.europa.eu/ghs\_stat\_ucdb2015mt\_r2019a.php
+	* License: CC-BY
+* ESRI’s Sentinel-2 10m Land Use/Land Cover;
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:23]: I think this is actually `Impact Observatory by National Geographic Society, Microsoft, and Esri`

And `Sentinel-2 10m Land Use/Land Cover (January 2022)` is the data layer (note with new sub-bullet).

File: sources/karmashapes.json [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            "context": "India", 
+            "source_link": "https://www.arcgis.com/apps/instant/media/index.html?appid=fc92d38533d440078f17678ebc20e8e2", 
+            "source_name": "ESRI’s Sentinel-2 10m Land Use/Land Cover", 
+            "source_note": ""
+        }, 
+        {
+            "context": "India", 
+            "source_link": "https://download.geoservice.dlr.de/WSF_EVO/", 
+            "source_name": "World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)",
```
Comment (nvkelso) [inline sources/karmashapes.json:47]: I think this source is `German Aerospace Center (DLR)` instead and `World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)` is the data layer.

File: sources/karmashapes_remarks.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+	* URL: https://www.arcgis.com/apps/instant/media/index.html?appid=fc92d38533d440078f17678ebc20e8e2
+	* License: CC-BY
+* World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)
+	* URL: https://download.geoservice.dlr.de/WSF\_EVO/
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:27]: Correct link to: `https://download.geoservice.dlr.de/WSF_EVO/`

File: sources/karmashapes_remarks.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+* Pradhan Mantri Gram Sadak Yojana (PMGSY);
+	* URL: https://geosadak-pmgsy.nic.in/opendata/
+	* License: CC-BY (equivalent)
+* Global Human Settlement Layer (GHSL);
+	* URL: https://ghsl.jrc.ec.europa.eu/ghs\_stat\_ucdb2015mt\_r2019a.php
+	* License: CC-BY
+* ESRI’s Sentinel-2 10m Land Use/Land Cover;
+	* URL: https://www.arcgis.com/apps/instant/media/index.html?appid=fc92d38533d440078f17678ebc20e8e2
+	* License: CC-BY
+* World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:26]: I think this source is `German Aerospace Center (DLR)` instead and `World Settlement Footprint (WSF) Evolution - Landsat-5/7 Dataset (WSFEVO)` is the data layer (as new sub-bullet)

File: sources/karmashapes_remarks.md [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+**Examples:** `ks:id` for Karmashapes identifier concordance and `ks:name` for place names. 
+
+Karmashapes is included in Who's On First's administrative data repo in India. Who's on First doesn't always use the Karmashapes geometry, so review that in `src:geom`. When we do, some Karmashapes features source from friendly CC0 or Public Domain licensed projects, and others from a stricter CC-BY project.
+
+Please note the following with regard to Quattroshapes features that require attribution:
+
+* National Geospatial Intelligence Agency (NGA)
+	* URL: https://geonames.nga.mil/geonames/GNSHome/welcome.html
+	* License: Public Domain
+* Socioeconomic Data and Applications Center (SECAC) / Center for International Earth Science Information Network (CIESIN);
```
Comment (nvkelso) [inline sources/karmashapes_remarks.md:14]: Don't need the `;` at the end of this line?

File: sources/karmashapes_remarks.md (hunk 12 above)
Comment (nvkelso) [reply sources/karmashapes_remarks.md:17]: And no `;` at end of line

File: sources/karmashapes_remarks.md (hunk 14 above)
Comment (nvkelso) [reply sources/karmashapes_remarks.md:20]: And no `;` at end of line

File: sources/karmashapes_remarks.md (hunk 17 above)
Comment (nvkelso) [reply sources/karmashapes_remarks.md:23]: And no `;` at end of the line

File: sources/karmashapes.json (hunk 2 above)
Comment (nvkelso) [reply sources/karmashapes.json:18]: (So never mind, pending seeing draft output of Markdown...)

**Comment by stepps00** [comment]
Overlooked this.. will handle comments so we can merge now that Karmashapes is in WOF.

File: sources/karmashapes_remarks.md (hunk 1 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:9]: Whoops!

File: sources/karmashapes_remarks.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (stepps00) [reply sources/karmashapes_remarks.md:11]: I added these two items (US agency reference, file being used) as a note in the source and markdown file.

File: sources/karmashapes_remarks.md (hunk 11 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:19]: Also added as a note.

File: sources/karmashapes_remarks.md (hunk 12 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:17]: Above, I see the data layer is called "Habitations".. are these the same data layers?

File: sources/karmashapes_remarks.md (hunk 13 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:22]: Added as a new note.

File: sources/karmashapes_remarks.md (hunk 15 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:25]: added

File: sources/karmashapes_remarks.md (hunk 16 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:28]: added

File: sources/karmashapes_remarks.md (hunk 9 above)
Comment (stepps00) [reply sources/karmashapes_remarks.md:12]: Fixed in most recent commits

**Comment by stepps00** [comment]
One comment about the PMGSY source, one comment that I can't seem to resolve (?), and a few merge conflicts..

@nvkelso want to give this one another look? If all looks good, I can resolve the conflicts to get this PR merged..

**Comment by nvkelso** [comment]
LGTM

**Review by nvkelso — APPROVED** [review]

**Comment by stepps00** [comment]
Unable to fix these merge conflicts, going to open up a separate PR to merge.
